Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -49,19 +52,66 @@
 
 void UnifiedPDFPlugin::createPDFDocument()
 {
-
+    auto dataProvider = adoptCF(CGDataProviderCreateWithCFData(m_data.get()));
+    m_pdfDocument = adoptCF(CGPDFDocumentCreateWithProvider(dataProvider.get()));
 }
 
 void UnifiedPDFPlugin::installPDFDocument()
 {
+    ASSERT(isMainRunLoop());
+
+    if (m_hasBeenDestroyed)
+        return;
+
+    if (!m_pdfDocument)
+        return;
+
+    if (!m_view)
+        return;
+
+}
+
+void UnifiedPDFPlugin::paint(GraphicsContext& context, const WebCore::IntRect& rect)
+{
+    ALWAYS_LOG_WITH_STREAM(stream << "UnifiedPDFPlugin::paint " << rect);
+    ASSERT(!context.paintingDisabled());
+
+    if (m_size.isEmpty())
+        return;
 
+    auto imageBuffer = ImageBuffer::create(m_size, RenderingPurpose::Unspecified, 1.0, DestinationColorSpace::SRGB(), PixelFormat::BGRA8);
```

Comment:
How is this code going to deal with large PDF files? The maximum size of an ImageBuffer is  4096 x 4096 pixels. Beyond that ImageBuffer is going to scale the context. So the PDF will be pixelated.